src/kerberos/client/generators.rs

```diff
@@ -345,7 +346,100 @@ pub fn generate_tgs_req(options: GenerateTgsReqOptions) -> Result<TgsReq> {
 #[derive(Debug)]
 pub struct ChecksumOptions {
     pub checksum_type: Vec<u8>,
-    pub checksum_value: Vec<u8>,
+    pub checksum_value: ChecksumValues,
+}
+
+#[derive(Debug)]
+pub struct ChecksumValues {
+    inner: Vec<u8>, // use named fields for future extensibility, this is a temporary solution
+}
+
+impl Default for ChecksumValues {
+    fn default() -> Self {
+        Self {
+            inner: AUTHENTICATOR_DEFAULT_CHECKSUM.to_vec(),
+        }
+    }
+}
+
+impl From<ChecksumValues> for Vec<u8> {
+    fn from(val: ChecksumValues) -> Self {
+        val.inner
+    }
+}
+
+impl From<[u8; 24]> for ChecksumValues {
+    fn from(bytes: [u8; 24]) -> Self {
+        ChecksumValues {
+            inner: Vec::from(bytes),
+        }
+    }
+}
+
+impl ChecksumValues {
+    pub(crate) fn set_flags(&mut self, flags: GssFlags) {
+        let flag_bits = flags.bits();
+        let flag_bytes = flag_bits.to_le_bytes();
+        self.inner[20..24].copy_from_slice(&flag_bytes);
+    }
+
+    pub(crate) fn get_inner(self) -> Vec<u8> {
```

Comment:
suggestion: Rename to `into_inner`.

rationale: The convention for methods deconstructing data into its underlying representation is `into_*`. [Rust API guidelines – C-CONV](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv).

Reply:
updated